api/src/client.rs

```diff
@@ -52,21 +52,26 @@ where
 /// on a given URL that returns nothing. Handles request
 /// building and response code checking.
 pub fn get_no_ret(url: &str, api_secret: Option<String>) -> Result<(), Error> {
-	send_request(build_request(url, "GET", api_secret, None)?)?;
+	send_request(build_request(url, "GET", api_secret, None)?, None)?;
 	Ok(())
 }
 
 /// Helper function to easily issue a HTTP POST request with the provided JSON
 /// object as body on a given URL that returns a JSON object. Handles request
 /// building, JSON serialization and deserialization, and response code
 /// checking.
-pub fn post<IN, OUT>(url: &str, api_secret: Option<String>, input: &IN) -> Result<OUT, Error>
+pub fn post<IN, OUT>(
+	url: &str,
+	api_secret: Option<String>,
+	input: &IN,
+	read_timeout: Option<u64>,
+) -> Result<OUT, Error>
```

Comment:
Instead of adding Option parameters to multiple methods, isn't it preferable to add a new timeout member,
and a method to change it from its default value of 20secs?

Reply:
I did it a bit more cleaner but just adding a method in api/client.rs and trigger a change of timeout in function of the api method used is a bit painful in the form it's built. If you have better idea, i'm open.
However these method are almost only used for cmd/client.rs